### ISSUE TITLE
How to use AWS Pinpoint with iOS10 UNUserNotificationCenter

### ISSUE THREAD

**Issue by helloniklas** [aws/aws-sdk-ios#577]
Since iOS10 has been out for a while now... how do I get the AWS Pinpoint to work with it.

Pinpoint only supports the old pinpoint.notificationManager.interceptDidReceiveRemoteNotification(userInfo, fetchCompletionHandler: completionHandler)

and not the new

 func userNotificationCenter(_ center: UNUserNotificationCenter, didReceive response: UNNotificationResponse, withCompletionHandler completionHandler: @escaping () -> Void)

**Comment by rcha** [comment]
Pinpoint will work with iOS10s UNUserNotificationFramework, you will need to call the interceptMethod inside the didReceive method callback.

**Comment by helloniklas** [comment]
@rcha That's what I'm asking though. What do I send to the intercept handler since it expects a completion handler with the typedef of ios9 methods. Seems the api needs to be updated to add an intercept call that matches the ios10 notification handler.

**Comment by helloniklas** [comment]
@rcha Just to clarify what the issue is:

```
    func userNotificationCenter(_ center: UNUserNotificationCenter, didReceive response: UNNotificationResponse, withCompletionHandler completionHandler: @escaping () -> Void) {
        
        AppDelegate.pinpoint.notificationManager.interceptDidReceiveRemoteNotification(response.notification.request.content.userInfo, fetchCompletionHandler: completionHandler)
        
    }

```

I can get the userInfo out and pass it in, but the completion handler parameters will give an error as it's of a different type.

**Comment by rcha** [comment]
@helloniklas Sorry for the confusion on the API, we will get a fix for it and remove the callback portion of the API since Pinpoint ignores the callback parameter. 

As a workaround for now you can call the intercept method with an empty callback:
In Swift
```swift
@available(iOS 10.0, *)
func userNotificationCenter(center: UNUserNotificationCenter, didReceiveNotificationResponse response: UNNotificationResponse, withCompletionHandler completionHandler: () -> Void) {
    AWSMobileClient.sharedInstance.pinpoint?.notificationManager.interceptDidReceiveRemoteNotification(response.notification.request.content.userInfo, fetchCompletionHandler: { (UIBackgroundFetchResult) in })
}
```

In Objc
```objective-c
- (void)userNotificationCenter:(UNUserNotificationCenter *)center
didReceiveNotificationResponse:(UNNotificationResponse *)response
         withCompletionHandler:(void (^)(void))completionHandler {
     [[[AWSMobileClient sharedInstance] pinpoint].notificationManager interceptDidReceiveRemoteNotification:response.notification.request.content.userInfo fetchCompletionHandler:^void(UIBackgroundFetchResult result) {}];
}
```
